### DISCUSSION TITLE
This browser requires https to use the web vault

### DISCUSSION THREAD

**Discussion by mathisokle** [dani-garcia/vaultwarden#1749] [Q&A]
### 
I get this error when I create an admin account in the web gui: "This browser requires https to use the web vault"

The Docker command:
 docker run -d --name bitwarden -p 8084:80 -p 8443:443 vaultwarden/server:latest

**Comment by RealOrangeOne** [comment]
Browsers require HTTPS for a lot of the security features.

This isn't an error, nor an issue with vaultwarden (it'd be bitwarden itself). Equally, It's definitely worth setting up HTTPS!

**Comment by BlackDex** [comment]
https://github.com/dani-garcia/vaultwarden/wiki/Enabling-HTTPS

**Comment by cat2devnull** [comment]
Just thought I would add a comment here because I hit the same issue myself and most of the documentation I found doesn't really talk about this. The main issue is that you can only access Vaultwarden via HTTPS but Vaultwarden like many projects don't want to have to trouble themselves with the difficulty of providing HTTPS, managing keys etc. They just want to focus on their actual app.
So what you have to do is forward all access through a reverse proxy (Nginx, Caddy, Swag, etc). The reverse proxy can then manage the HTTPS side of things.
If you are wanting to self host a docker for Vaultwarden then you should have the technical skills to setup a docker for a reverse proxy in combination with DDNS. There are many great videos on YouTube on how to setup these extra components.

The issue then arrises, how do I force all traffic to my Vaultwarden server through the reverse proxy?

This is where DNS comes to help. For external access you just set your CNAME for your Vaultwarden instance to the external address of your network using DDNS. Make sure your firewall has NAT configured to forward traffic to your reverse proxy and you're good to go. Again see the many YouTube videos on this.

Where I got stuck was how to get my local LAN traffic to use HTTPS. I was getting the message "This browser requires https to use the web vault" when ever I tried to access the server via a web browser. In the end it was simple. I just had to update my local DNS server to point at the reverse proxy rather than directly at the Vaultwarden server.

I've attached a little diagram to make it more obvious.
![Screenshot 2023-01-17 at 9 49 27 am](https://user-images.githubusercontent.com/69185925/212776463-7eccf0fd-21a1-4136-8cd3-4259c9f6ff30.png)

**Comment by MushyMiddle** [comment]
I think this is mostly a doc issue.  Most of the wiki examples in terms of setup refer to port 80, but if HTTPS **must** be used (as seems to be the case), then examples shouldn't lead you to think that port 80 is OK.  I know it's complicated to address HTTPS before getting basic functionality working, but it's a bit silly to get things running in an unusable state, and then have to fall back and set up HTTPS.  I would far prefer that port 80 be left out of the docs/setup entirely - it's a mystery why it's there in the first place.

**Reply by BlackDex** [reply]
It's there because reverse proxies are most of the time used in front of Vaultwarden. Also, most reverse proxies are much better in handling all the different clients which could use broken http calls in some way.

And, before we were able to use websockets on the same port it was mandatory to use a reverse proxy.

Also, i think the TLS support in Rust is maybe not as robust as all the reverse proxies have. So that would still be an argument.

Same goes for certificate handling with LetsEncrypt for example. There isn't anything (right now) in Vaultwarden to handle that.

**Comment by MushyMiddle** [comment]
OK, just wish I had seen this: https://github.com/bitwarden/web/issues/254 (see last section) before I decided to try the HTTP route.
